File: Examples/GroupDocs.Watermark.Examples.VBasic/Images.vb
Imports GroupDocs.Watermark.Images
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks

Namespace GroupDocs.Watermark.Examples.VBasic
    Public NotInheritable Class Images
        Private Sub New()
        End Sub
        ' initialize file path
        'ExStart:SourceImageFilePath
        Private Const FilePath As String = "Images/sample.jpg"
        'ExEnd:SourceImageFilePath

        ''' <summary>
        ''' Adds watermark to an image
        ''' </summary> 
        Public Shared Sub AddWatermark()
            Try
                'ExStart:AddWatermarkToImage
                Using doc As MultiframeImageDocument = Document.Load(Of MultiframeImageDocument)(FilePath)
                    ' Initialize text or image watermark
                    Dim watermark As New TextWatermark("Test watermark", New Font("Arial", 19))

                    ' Add watermark to the first frame
                    doc.Frames(0).AddWatermark(watermark)

                    doc.Save()
                End Using
                'ExEnd:AddWatermarkToImage
            Catch exp As Exception
                Console.Write(exp.Message)
            End Try
        End Sub

    End Class
End Namespace

File: Examples/GroupDocs.Watermark.Examples.CSharp/Images.cs
using GroupDocs.Watermark.Images;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace GroupDocs.Watermark.Examples.CSharp
{
    public static class Images
    {
        // initialize file path
        //ExStart:SourceImageFilePath
        private const string FilePath = "Images/sample.jpg";
        //ExEnd:SourceImageFilePath

        /// <summary>
        /// Adds watermark to an image
        /// </summary> 
        public static void AddWatermark()
        {
            try
            {
                //ExStart:AddWatermarkToImage
                using (MultiframeImageDocument doc = Document.Load<MultiframeImageDocument>(FilePath))
                {
                    // Initialize text or image watermark
                    TextWatermark watermark = new TextWatermark("Test watermark", new Font("Arial", 19));

                    // Add watermark to the first frame
                    doc.Frames[0].AddWatermark(watermark);

                    doc.Save();
                }
                //ExEnd:AddWatermarkToImage
            }
            catch (Exception exp)
            {
                Console.Write(exp.Message);
            }
        }

    }
}
